### DISCUSSION TITLE
route_bundle() with bboxes not working

### DISCUSSION THREAD

**Discussion by pvasired** [gdsfactory/gdsfactory#2980] [Q&A] [answered]
I am trying to use route_bundle() with bboxes a list of obstacles to avoid, but the traces are not avoiding the obstacle and instead going straight through them. The router with route_bundle() is working properly, but there is no difference in routing when I add this obstacle, even though the obstacle is placed into the .gds file. Any advice?

Image of the issue is linked in the code below as well. If there is any more information required to answer this, please let me know and I will provide it.

```
cross_section = gf.cross_section.cross_section(width=trace_width, layer=(layer_number, 0))
obstacle = gf.Component("Obstacle")
obstacle.add_polygon([(550, 300), (850, 300), (850, 400), (550, 400)], layer=(layer_number, 0))
gdspath.add_ref(obstacle)

# Debugging: Print obstacle details
bbox_obstacle = obstacle.bbox()
print("Obstacle Bounding Box:", bbox_obstacle)

# gdspath is loaded from a .gds file as gdspath = gf.read.import_gds(filename) and then ports are added manually
![obstacle_collision](https://github.com/user-attachments/assets/aac1e42e-88b7-470d-a615-56868cf23336)

routing.route_bundle(gdspath, ports1=[gdspath.ports[f"Electrode {i}"] for i in range(len(sorted_inds), len(sorted_inds) + len(sorted_inds2))], 
                     ports2=[gdspath.ports[f"Top Pad {i}"] for i in range(len(sorted_inds), len(sorted_inds) + len(sorted_inds2))], cross_section=cross_section,
                     separation=trace_width, bboxes=[bbox_obstacle])

gdspath.write_gds(filename)
```

**Comment by sebastian-goeldi** [comment, marked as answer]
Hello,

This has not been implemented on the kfactory side, yet. Not sure about gdsfactory, but kfactory mentions this in the docs for the function: https://github.com/gdsfactory/kfactory/blob/main/src/kfactory/routing/optical.py#L534-L535

The reason for not implementing it, yet, is that it's not that straight forward to implement a good solution.

Same for waypoints by the way.

Best,
Sebastian

**Reply by pvasired** [reply]
Thanks for the information!
